velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -209,4 +209,93 @@ struct LastDayFunction {
   }
 };
 
+template <typename T>
+struct MonthsBetweenFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Timestamp>* /*timestamp1*/,
+      const arg_type<Timestamp>* /*timestamp2*/,
+      const bool /*roundOff*/) {
+    timeZone_ = getTimeZoneFromConfig(config);
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getYear(const std::tm& time) {
+    return 1900 + time.tm_year;
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getMonth(const std::tm& time) {
+    return 1 + time.tm_mon;
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getDay(const std::tm& time) {
+    return time.tm_mday;
+  }
+
+  FOLLY_ALWAYS_INLINE std::tuple<int32_t, int32_t, int32_t, int32_t> splitDate(
+      Timestamp timestamp) {
+    std::tuple<int32_t, int32_t, int32_t, int32_t> result;
+    int32_t year = getYear(getDateTime(timestamp, this->timeZone_));
+    int32_t monthInYear = getMonth(getDateTime(timestamp, this->timeZone_));
+    int32_t dayInMonth = getDay(getDateTime(timestamp, this->timeZone_));
+    int32_t daysToMonthEnd = util::getMaxDayOfMonth(year, monthInYear) -
+        getDay(getDateTime(timestamp, this->timeZone_));
+    result = std::make_tuple(year, monthInYear, dayInMonth, daysToMonthEnd);
+    return result;
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getSecondsInDay(
+      Timestamp timestamp,
+      int32_t year,
+      int32_t month,
+      int32_t day) {
+    const date::time_zone* timeZone = this->timeZone_;
+    Timestamp t1 =
+        util::fromDatetime(util::daysSinceEpochFromDate(year, month, day), 0);
+    if (timeZone != nullptr) {
+      t1.toGMT(*timeZone);
+    }
+    return (timestamp.toMicros() - t1.toMicros()) / 1000000;
```

Comment:
Maybe use existing static variables in Timestamp.h to replace 1000000.
E.g., kMicrosecondsInMillisecond * kMillisecondsInSecond.

Reply:
done